rp2040-hal/src/timer.rs

```diff
@@ -103,26 +103,26 @@ impl Timer {
 /// Delay implementation
 pub struct CountDown<'timer> {
     timer: &'timer Timer,
-    period: embedded_time::duration::Microseconds<u64>,
+    period: MicrosDurationU64,
     next_end: Option<u64>,
 }
 
 impl embedded_hal::timer::CountDown for CountDown<'_> {
-    type Time = embedded_time::duration::Microseconds<u64>;
+    type Time = MicrosDurationU64;
 
     fn start<T>(&mut self, count: T)
     where
         T: Into<Self::Time>,
     {
         self.period = count.into();
-        self.next_end = Some(self.timer.get_counter().wrapping_add(self.period.0));
+        self.next_end = Some(self.timer.get_counter().wrapping_add(self.period.ticks()));
```

Comment:
Isnt it nicer to use `to_micros()` to make it more explicit especially if someone might want to change `self.period` to another base?

Reply:
You're probably right. Although, something feels wrong.
The assumption that the counter increments every us only holds if the user configures the ref clk of the watchdog to be at that frequency.
If that's not the case most of our code would be off.

Reply:
As far as I can see in the reference manual the watchdog and timer tick interval is based on ref_clock but internally divided to always be as close to 1us per tick.

Reply:
The division is controlled by [`init_init_clocks_and_plls`](https://github.com/rp-rs/rp-hal/blob/main/rp2040-hal/src/clocks/mod.rs#L332) which sets the `CYCLES` field of the `TICK` register described in [4.7.6](https://datasheets.raspberrypi.com/rp2040/rp2040-datasheet.pdf#reg-watchdog-TICK).

So while unlikely, it is possible for a user to use a different value there.

Note:
I noticed that the `CYCLE` field is 9bits (8:0) wide while our method only takes and 8bit value.
I'll have to fix that, and introduce an extra `watchdog::Error::BadArgument` to `watchdog::Watchdog::enable_tick_generation`.